x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/BlockUtils.java

```diff
@@ -52,7 +53,12 @@ public BuilderWrapper(Block.Builder builder, Consumer<Object> append) {
         }
 
         public void accept(Object object) {
-            append.accept(object);
+            try {
+                append.accept(object);
+            } catch (CircuitBreakingException e) {
+                close();
+                throw e;
+            }
```

Comment:
why is this needed?

Reply:
Tests that use the Cranky CBE throw a CBE one every 20 times. This code path was not taking into account CBEs and memory was not released when a CBE exception was hit, thus failing tests.